tensorrt_llm/_torch/custom_ops/torch_custom_ops.py

```diff
@@ -1035,6 +1042,235 @@ def _(
+    )
+
+    def __init__(self):
+        super().__init__()
+
+    def get_valid_tactics(
+        self,
+        inputs: List[torch.Tensor],
+        profile: OptimizationProfile,
+        **kwargs,
+    ) -> List[int]:
+        m = inputs[0].shape[0]
+        n = inputs[1].shape[0]
+        k = inputs[0].shape[1]
+        # Note: the input tensor use uint8 to store fp4, so the real_k is k * 2
+        real_k = k * 2
+        batch_size = 1
+        # m,k
+        a_major = "k"
+        # n, k
+        b_major = "k"
+        # m, n
+        c_major = "n"
+        sf_vec_size = 16
+
+        # full shamoo
+        mma_tiler_mn_candi = [(256, 128), (128, 128), (128, 256), (256, 256),
+                              (256, 64), (128, 64)]
+        cluster_shape_mn_candi = [(1, 1), (1, 2), (1, 4), (2, 1), (2, 2),
+                                  (2, 4), (4, 1), (4, 2), (4, 4)]
+        return [
+            (mma_tiler_mn, cluster_shape_mn)
+            for mma_tiler_mn in mma_tiler_mn_candi
+            for cluster_shape_mn in cluster_shape_mn_candi
+            if Sm100BlockScaledPersistentDenseGemmKernel.can_implement(
+                cutlass.Float4E2M1FN,  # ab_dtype,
+                cutlass.Float8E4M3FN,  # sf_dtype
+                sf_vec_size,  # sf_vec_size,
+                cutlass.BFloat16,  # c_dtype,
+                mma_tiler_mn,
+                cluster_shape_mn,
+                m,
+                n,
+                real_k,
+                batch_size,
+                a_major,
+                b_major,
+                c_major,
+            )
+        ]
+
+    def forward(
+        self,
+        inputs: List[torch.Tensor],
+        tactic,
+    ) -> torch.Tensor:
+        """Performs fp8 blockwise (deepgemm like) operation using CuTe DSL.
+        :param a (inputs[0]): Input tensor of shape (m, k)
+        :type a: torch.Tensor, type: fp4
+        :param b (inputs[1]): Weight tensor of shape (n, k)
+        :type b: torch.Tensor, type: fp4
+        :param a_sf (inputs[2]): Input scale tensor of shape (k//16, m).
+        :type a_sf: torch.Tensor, type: fp8
+        :param b_sf (inputs[3]): Weight scale tensor of shape (n, k//16)
+        :type b_sf: torch.Tensor, type: fp8
+        :return: Output tensor of shape (m, n)
+        :rtype: torch.Tensor, type: bf16
+        """
+        sf_vec_size = 16
+
+        if isinstance(tactic, tuple):
+            mma_tiler_mn, cluster_shape_mn = tactic
+        else:
+            # fallback to default tactic
+            mma_tiler_mn, cluster_shape_mn = [
+                (128, 128),
+                (1, 1),
+            ]
+
+        a_tensor, b_tensor, a_sf_tensor, b_sf_tensor, alpha, output_dtype = inputs
+        assert output_dtype == torch.bfloat16
+        m, k, n = a_tensor.shape[0], a_tensor.shape[1], b_tensor.shape[0]
+        c_tensor = torch.empty(*(m, n), dtype=output_dtype, device="cuda")
+
+        real_k = k * 2
+        sf_m = pad_up(m, 128)
```

Comment:
There is no actual padding on the `a_sf` and `b_sf` tensors inside this function right? Do we need some assertion of their shapes must be already padded, otherwise will there be illegal memory access?